packages/next/src/client/components/react-dev-overlay/internal/helpers/use-error-handler.ts

```diff
@@ -2,29 +2,40 @@ import { useEffect } from 'react'
 import { isHydrationError } from '../../../is-hydration-error'
 import { attachHydrationErrorState } from './attach-hydration-error-state'
 import { isNextRouterError } from '../../../is-next-router-error'
+import { storeHydrationErrorStateFromConsoleArgs } from './hydration-error-info'
+import { formatConsoleArgs } from '../../../../lib/console'
+import isError from '../../../../../lib/is-error'
 
 export type ErrorHandler = (error: Error) => void
 
-let hasHydrationError = false
 const errorQueue: Array<Error> = []
 const errorHandlers: Array<ErrorHandler> = []
 const rejectionQueue: Array<Error> = []
 const rejectionHandlers: Array<ErrorHandler> = []
 
-export function handleClientError(error: unknown) {
-  if (!error || !(error instanceof Error) || typeof error.stack !== 'string') {
-    // A non-error was thrown, we don't have anything to show. :-(
-    return
+export function handleClientError(
+  originError: unknown,
+  consoleErrorArgs: any[]
+) {
+  let error: Error
+  if (!originError || !isError(originError)) {
```

Comment:
I think we need to make sure that this is also skipping `instanceof Error` checks as I don't think `isError` is sufficient. ie 

```tsx

export default function Page() {
  console.error(new Error('kaboom'))
  return <h1>Home</h1>
}

```

gives this:

![CleanShot 2024-10-17 at 07 33 36@2x](https://github.com/user-attachments/assets/5e4ba947-05b0-4e30-8a71-3ab845440e6c)